public/app/features/scopes/internal/ScopesDashboardsScene.tsx

```diff
@@ -89,26 +94,50 @@ export class ScopesDashboardsScene extends SceneObjectBase<ScopesDashboardsScene
 
     this.setState({ isLoading: true });
 
-    const dashboards = await fetchSuggestedDashboards(scopeNames);
+    const dashboards = await fetchDashboards(scopeNames);
+    const folders = groupDashboards(dashboards);
+    const filteredFolders = filterFolders(folders, this.state.searchQuery);
 
     this.setState({
       dashboards,
-      filteredDashboards: this.filterDashboards(dashboards, this.state.searchQuery),
+      folders,
+      filteredFolders,
       forScopeNames: scopeNames,
       isLoading: false,
       scopesSelected: scopeNames.length > 0,
     });
   }
 
   public changeSearchQuery(searchQuery: string) {
```

Comment:
I suggest debouncing this, otherwise this can create a lot of unnecessary re-renders.

Reply:
Implemented debounce